lib/service/supervisor.go

```diff
@@ -593,7 +593,7 @@ func (l *LocalService) Name() string {
 }
 
 // Func is a service function
-type Func func() error
+type Func = func() error
```

Comment:
This avoids the circular dependency with lib/autoupdate/adaptor.

I did a first version of this PR with a supervisor refactor but this grew out of control too quickly. I think this part of Teleport would benefit from a bit of cleaning (the supervisor and the process are too linked together, supervisor depends on hardcoded function names and specific Teleport events). However this is too much work for me given the timelines we have for canary updates.